packages/types/types/index.d.ts

```diff
@@ -74,6 +85,24 @@ export interface HideProps extends BoxProps {
   print?: boolean
 }
 
+export interface IconProps
+  extends ColorProps,
+  SizeProps {
+  name?: string
+  title?: string
+  titleId?: string
+  desc?: string
+  descId?: string
+}
+
+export interface IconButtonProps extends ButtonProps {
+  icon?: React.ComponentElement<IconProps, any>
+  onClick?: Function
```

Comment:
I think `onClick` could probably go on `ButtonProps` instead since it's needed there too

Reply:
Makes sense, moved into `ButtonProps` and also added a more specific type (`React.MouseEventHandler<HTMLElement>`) based on linter feedback!